README.md

```diff
@@ -1,38 +1,148 @@
+And once a choice is made, it is awkward to adjust in CAKE.
 
-_For Testers, Jan 2022:_ For those people running OpenWrt snapshot builds,
-a patch is required for Lua Lanes.
-Details can be found here:
-[https://github.com/sqm-autorate/sqm-autorate/issues/32#issuecomment-1002584519](https://github.com/sqm-autorate/sqm-autorate/issues/32#issuecomment-1002584519)
+Remember that while network speed is nice, beyond a necessary base speed, latency is what makes a network good for gaming, streaming, video calls, and general use.
+And this is multiplied when several people share a common router.
+Particularly if, for example, one is torrenting large files, while another is gaming, and another is in that career changing conference call.
+
+![Image of Bandwidth Compromise](.readme/Bandwidth-Compromise.png)
+
+Image credit [Lynxthecat](https://github.com/lynxthecat)
+
+### What does _sqm-autorate_ do?
+
+#### In consise terms:
+
+_sqm-autorate_ monitors the current latency and current demand for bandwidth as seen on the router,
+then actively adjusts the CAKE speed to get the best current compromise.
+And it does it independently for both uploads and downloads.
+
+#### In more and still incomplete detail:
+
+_sqm-autorate_ measures the [Round Trip Time (RTT)](https://en.wikipedia.org/wiki/Round-trip_delay)
+to a number of [ICMP reflectors](https://en.wikipedia.org/wiki/Internet_Control_Message_Protocol)
+and splits these into the upload and download components.
+It uses some [cool maths](https://en.wikipedia.org/wiki/Moving_average#Exponential_moving_average)
+to keep a longer term _baseline_ RTT and a fast moving _recent_ RTT.
+* If the recent RTT is higher than the baseline, then the difference is the _delay_.
+* If the recent RTT is lower than the baseline, then the baseline is reset to to the recent,
+because there is no such thing as negative delay on a network.
+
+If there is high bandwidth utilisation (aka. load), with low delay, then the CAKE speed is adjusted up.
+If there is low load and high delay, then the CAKE speed is adjusted down.
+And all this is done separately for upload and download speeds, twice each second.
+
+There's much more going on, please ask in the discussions, though we suggest to first have a look at the rest of this document and then at the code.
+
+### What happens when _sqm-autorate_ is running?
+_sqm-autorate_ starts out with the CAKE speeds set to a chosen minimum acceptable speed.
+
+When network activity occurrs that uses 80% or more of the current speed and the _delay_ is less than 15 millisconds (ms), then the speed is increased.
+At lower speeds, the increase can be quite large.
+As the speeds get towards the chosen target speed, the size of the increase is reduced.
+_sqm-autorate_ will set the speed to rates higher than the chosen target if it does not see any delay.
+It can even set the speeds higher than the maximum the connection is physically capable of.
```

Comment:
Maybe something like this is clearer?
```suggestion
It will increase the rate while load is high and delays are low until delays start rising, indicating that it's now at the physical limit of the connection, and then back off until delays lower again
```

Reply:
I think that is covered in the following paragraph